lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DriverRuntimeRestartManager.java

```diff
@@ -58,9 +59,9 @@
 
   /**
    * Gets the sets of alive and failed evaluators based on the runtime implementation.
-   * @return EvaluatorRestartInfo, which encapsulates the alive and failed set of evaluator IDs.
+   * @return A map which encapsulates the states of previous evaluators.
    */
-  EvaluatorRestartInfo getAliveAndFailedEvaluators();
+  Map<String, EvaluatorRestartInfo> getPreviousEvaluators();
```

Comment:
Access to this Map happens only in `DriverRestartManager`, where it is used in the various `get()` methods. E.g. the `getFailedEvaluators()` call. Would it make sense to capture all that structure in its own collection object, e.g. `RestartEvaluators` which has access methods for the various conditions?